DTS-STN/canadian-dental-care-plan
# Adding server modules (service, repository, DTO, entity, mapper, etc.) for applicant

### Description
These modules are intended to replace the existing `personal-information-service`, `personal-information-service-mappers`, and `personal-information-service-schemas`.

This PR can be broken down into smaller PRs upon request.

### Screenshots (if applicable)
Request:
![image](https://github.com/user-attachments/assets/60d4e480-081c-4f76-90ba-2d596fcd5446)

Result:
![image](https://github.com/user-attachments/assets/b80199d6-d2b3-42e5-a397-615e491d62a9)

### Checklist

### Test Instructions
Update some route to use the `serviceProvider` to call the `findClientNumberBySin` from the applicant service. The mock should be hit (if it's enabled).

### Additional Notes
References to existing personal information modules will be changed and removed in future PRs.
